app/helpers/sessions_helper.rb

```diff
@@ -36,4 +36,17 @@ def forget user
     cookies.delete :user_id
     cookies.delete :remember_token
   end
+
+  def current_user? user
+    user && user == current_user
+  end
+
+  def redirect_back_or default
+    redirect_to session[:forwarding_url] || default
+    session.delete :forwarding_url
```

Comment:
Thử xem như này có chạy ok không em 
```suggestion
    redirect_to  session.delete(:forwarding_url) || default
```

Reply:
Em thử thì nó lỗi anh ạ